RatS/filmtipset/filmtipset_ratings_parser.py

```diff
@@ -0,0 +1,82 @@
+import os
+import datetime
+import time
+import fileinput
+import re
+from shutil import copyfile
+from RatS.base.base_ratings_downloader import RatingsDownloader
+from RatS.filmtipset.filmtipset_site import Filmtipset
+from RatS.utils import file_impex
+
+# Request download form
+# https://www.filmtipset.se/installningar
+# <form method="post" autocomplete="off">
+#     <select id="export-type" name="export-type" placeholder="Välj dataslag" required="">
+#     <option value="betyg">Filmbetyg
+#     </option><option value="kommentarer">Filmkommentarer
+#     </option><option value="recensioner">Recensioner
+#     </option></select><br>
+#     <input type="submit" id="submit-export" name="submit-export" value="Exportera data" class="button"><br>
+# </form>
+
+class FilmtipsetRatingsParser(RatingsDownloader):
+    def __init__(self, args):
+        super(FilmtipsetRatingsParser, self).__init__(Filmtipset(args), args)
+        # TODO: due to timezone and potential racecondition, it's probably more safe to
+        #       find/search file in download folder
+        date = datetime.datetime.fromtimestamp(time.time()).strftime('%Y-%m-%d')
+        self.downloaded_file_name = 'ft_betyg_{}.csv'.format(date)
+        self.csv_delimiter = ';'
+
+    # TODO remove when filmtipset has cleaned up thier "csv" file
+    @staticmethod
+    def _repair_line(line):
+        output = re.search(r'([0-9]{4}-[0-9]{2}-[0-9]{2}),(.*)(;[\-0-9]*;[1-5])$', line)
+        if output is None:
+            return line
+        group2 = output.group(2).replace('"', '""')
+        newline = '{};"{}"{}\n'.format(output.group(1), group2, output.group(3))
+        return newline
+
+    @staticmethod
+    def _repair_csv_file(filepath):
+        copyfile(filepath, filepath + '.bak')
+        for line in fileinput.input(filepath, inplace=True): 
+            print(FilmtipsetRatingsParser._repair_line(line),end="")
+
+    def _parse_ratings(self):
+        self._download_ratings_csv()
+        self._rename_csv_file(self.downloaded_file_name)
+        filepath = os.path.join(self.exports_folder, self.csv_filename)
+        self._repair_csv_file(filepath)
+        self.movies = self._parse_movies_from_csv(filepath)
+
+    def _call_download_url(self):
+        self.site.browser.find_element_by_xpath('//input[@id="submit-export"]').click()
+        self.site.browser.get('https://www.filmtipset.se/export-data.php?q=')
+
+    #VoteDate;MovieTitle;IMDB;Score
+    def _convert_csv_row_to_movie(self, headers, row):
+        movie = dict()
+        movie['title'] = row[headers.index("MovieTitle")]
+        movie[self.site.site_name.lower()] = dict()
+        filmtipset_rating = int(row[headers.index("Score")])
+        movie[self.site.site_name.lower()]['my_rating'] = filmtipset_rating
```

Comment:
I am not sure if I should keep this as the original 1 - 5 grade or not.

Reply:
So far, I always translated the site's rating scale to a unified 1-10 scale. Only this way, we can be sure that a 3 in filmtipset is the same as a 3 in Trakt.

Reply:
```suggestion
        movie[self.site.site_name.lower()]['my_rating'] = filmtipset_rating * 2
```